Automattic/newspack-theme
# Add child theme to replace style 2

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds a child theme for Style 2, named Newspack Nelson (after [Nelson Poynter](https://href.li/?https://en.wikipedia.org/wiki/Nelson_Poynter); credit to @philipjohn for the name suggestion! 

It also updates the main Newspack theme to use the Sass files from the child theme to build the CSS for style 2. 

There is still some duplicate code: `newspack-theme/inc/color-patterns.php` and `newspack-theme/inc/typography.php` contain a lot of the same code as `newspack-nelson/inc/child-color-patterns.php` and `newspack-nelson/inc/child-typography.php`; this will likely need to stay this way until the style pack code is fully removed from the main theme.

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`; confirm that there are no errors.
2. If you're symlinking the theme on your dev site, create a second symlink for the new Newspack Nelson theme.
3. Navigate to WP Admin > Themes, and confirm the presence of Newspack Nelson. 
4. Try switching to the theme and confirm it loads right; looks like style 2.
5. Confirm the editor is loading styles correctly.
6. Try setting custom fonts and colours, and confirm they're loading on the front-end and in the editor.
7. Confirm that there is no "Style Packs" option in the Customizer.
8. Switch back to Newspack theme.
9. Confirm that the "Style Packs" option is back in the Customizer.
10. Switch to "Style 2"; confirm that it still works. 
11. Run `npm run release:archive`; confirm that the /releases directory contains `newspack-theme.zip`, `newspack-sacha.zip`, `newspack-scott.zip`, and now `newspack-nelson.zip`.
12. As an extra precaution, switch back to `master`. Move your generated zip files out of the releases directory, and re-run `npm run release:archive`, and run a file compare of the Newspack theme, to confirm that the generated CSS for Style 2 is the same, minus the header comments. 

### Other information:
